lib/plugins/aws/package/compile/events/mq.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+class AwsCompileMQEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileMQEvents.bind(this),
+    };
+
+    const secretsManagerArnRegex =
+      'arn:[a-z-]+:secretsmanager:[a-z0-9-]+:\\d+:secret:[A-Za-z0-9/_+=.@-]+';
+
+    const mqBrokerArnRegex =
+      'arn:[a-z-]+:mq:[a-z0-9-]+:\\d+:broker:[A-Za-z0-9/_+=.@-]+:b-[a-z0-9-]+';
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'mq', {
+      type: 'object',
+      properties: {
+        arn: {
+          anyOf: [
+            { type: 'string', pattern: mqBrokerArnRegex },
+            { $ref: '#/definitions/awsCfImport' },
+            { $ref: '#/definitions/awsCfRef' },
+          ],
+        },
+        accessConfigurations: {
+          type: 'object',
+          minProperties: 1,
+          maxProperties: 1,
```

Comment:
It's required here to pass only one configuration of `basicAuth` type? If that's the case, I wonder if it wouldn't be better to just go with `accessConfiguration: <arn>`. What do you think?

Reply:
Yeah we could definitely do that and I think it'd cut down on configuration.  I guess what are the chances other accessConfigurations would be possible in the future and would it cause pain for those if we needed to migrate the schema?  That is if we needed to add another accessConfiguration in the future would we want to support both accessConfiguration implying `basicAuth` as well as a object of values later on?  I'm not sure if this is leaving too much to a what-if scenario.

Reply:
I'm also a bit torn on this one for this very reason - potentially supporting more options in the future. We had similar discussion with @medikoo on PR about EFS (https://github.com/serverless/serverless/pull/8042) where CF expects array of `fileSystemConfigs`, but supports at most one. We went with `fileSystemConfig` on Framework side to simplify configuration for users and we will still have a reasonable migration path in case it's needed (we still can introduce `accessConfigurations` and deprecate `accessConfiguration` in such case). What do you both think @lewgordon @medikoo ?

Reply:
Hmm I wonder if `accessConfigurations` and `accessConfiguration` is too easy to typo.  What about putting `basicAuth` as a top level property and migrate to `accessConfigurations` in the future if need be?

Reply:
I would probably go with just `authSecretArn: <arn>` (better name welcome) directly in event config, and if for some reason AWS will extend authentication options. Then we may introduce `accessConfigurations`.

Still here situation is that AWS uses same resource for many event types, so fact that it's nested in array on AWS side, doesn't necessarily mean that they envision adding support for more types in case of MQ.

Reply:
I think the suggestion to have it as a property directly in event config is the best approach for now and even if we need to change it at some point in the future, it's not going to be a very big migration. :+1:

Reply:
> I would probably go with just `authSecretArn: <arn>` (better name welcome)...

Any thoughts on `basicAuthArn: <arn>`? @medikoo @pgrzesik

Reply:
@lewgordon Looks great to me 👍